add timing VS lumi (for HLT monitoring)

mtosi opened the pull request:
back port of https://github.com/cms-sw/cmssw/pull/14937

gregor-mittag committed:
Added plugin to create ideal tracker alignment payloads.

 - takes the geometry stored in the global tag
 - allows to align to the alignment payloads in the global tag
 - has option to skip subdetectors when aligning the ideal records to the
   given global tag
 - the ideal surface deformation payload is empty, but if one aligns to a
   none-ideal global tag it might contain elements
 - allows for creation of reference TrackerAlignmentRcd:
     If the corresponding flag is set to True, the created payload is just
     a vector of zeros, i.e. all modules are at the origin.

gregor-mittag committed:
Allow zero material in the construction of the reference trajectories.

 - configuration option 'AllowZeroMaterial' added to TrajectoryFactory
 - this option is needed if the material description is missing for parts of the
   tracker and if the material should be properly handled for all other parts

Commit:
adding new draft version of the ecal trigger alcareco

gregor-mittag committed:
Reduce PCL Mille binary size by dropping fixed alignables.

gregor-mittag committed:
Make convenience methods in 'AlignableTracker' geometry-independent.

gregor-mittag committed:
Make AlignableObjectId utilities static functions.

gregor-mittag committed:
Make 'TrackerScenarioBuilder' geometry-independent.

mtosi committed:
add hit eff vs lumi (estimated by pixel clusters)

tstreble committed:
Changed L1EG HoE implementation - adjusted and resolved conflict from PR cms-l1t-offline #353

Conflicts:
	L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2EGammaAlgorithmFirmwareImp1.cc

mtosi committed:
add lumi VS LS plot

get 81x GetLumi.h

mtosi committed:
add timing vs lumi plots

cmsbuild commented:
A new Pull Request was created by  @mtosi (mia tosi) for CMSSW_8_0_X.

It involves the following packages:

DQM/HLTEvF
DQM/TrackingMonitor
HLTrigger/Timer


@perrotta, @cmsbuild, @dmitrijus, @Martin-Grunewald, @fwyzard, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@makortel, @Martin-Grunewald, @threus, @fioriNTU this is something you requested to watch as well.
@slava77, @degano, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

fwyzard commented on DQM/HLTEvF/plugins/LumiMonitor.cc:
```diff
@@ -0,0 +1,212 @@
+  
+{
+
+  folderName_ = conf_.getParameter<std::string>("FolderName");
```
can you move `folderName_`, `pixelClustersToken_` and `lumiscalersToken_` to the initialisation as well ?

fwyzard commented on DQM/HLTEvF/plugins/LumiMonitor.cc:
```diff
@@ -0,0 +1,212 @@
+
+  folderName_ = conf_.getParameter<std::string>("FolderName");
+
+  pixelClustersToken_ = mayConsume<edmNew::DetSetVector<SiPixelCluster> >(conf_.getParameter<edm::InputTag>("pixelClusters"));
```
why `mayConsume` rather than `consumes` ?

fwyzard commented on DQM/HLTEvF/plugins/LumiMonitor.cc:
```diff
@@ -0,0 +1,212 @@
+  pixelClustersToken_ = mayConsume<edmNew::DetSetVector<SiPixelCluster> >(conf_.getParameter<edm::InputTag>("pixelClusters"));
+  lumiscalersToken_   = mayConsume<LumiScalersCollection>(conf_.getParameter<edm::InputTag>("scalers"));
+
+  numberOfPixelClustersVsLS_   = NULL;
```
could you use `nullptr` instead of `NULL` for new code ?

fwyzard commented on DQM/HLTEvF/plugins/LumiMonitor.h:
```diff
@@ -0,0 +1,74 @@
+{
+public:
+  LumiMonitor( const edm::ParameterSet& );
+  ~LumiMonitor();
```
I would suggest
```
~LumiMonitor() = default;
```

and to remove the empty implementation from the .cc file.

fwyzard commented on DQM/HLTEvF/plugins/LumiMonitor.cc:
```diff
@@ -0,0 +1,212 @@
+
+}
+
+void LumiMonitor::fillHistoPSetDescription(edm::ParameterSetDescription & pset)
```
Is there a particular reason to modify an existing `edm::ParameterSetDescription`, rather than creating a new one ?
This way you could slightly simplify the code that uses `fillHistoPSetDescription`.

fwyzard commented on DQM/HLTEvF/plugins/LumiMonitor.cc:
```diff
@@ -0,0 +1,212 @@
+  pset.add<double>( "xmax" );
+}
+
+void LumiMonitor::fillHistoLSPSetDescription(edm::ParameterSetDescription & pset)
```
same comment as for `fillHistoPSetDescription`.

fwyzard commented on DQM/HLTEvF/plugins/LumiMonitor.h:
```diff
@@ -0,0 +1,74 @@
+
+private:
+
+  void bookHistograms();
```
mhm, what do you need this for ?

fwyzard commented on DQM/HLTEvF/plugins/LumiMonitor.h:
```diff
@@ -0,0 +1,74 @@
+
+  float lumi_factor_per_bx_;
+
+  unsigned long long m_cacheID_;
```
I think this is unused ?

fwyzard commented on DQM/TrackingMonitor/interface/GetLumi.h:
```diff
@@ -24,15 +24,28 @@ class GetLumi
-  static constexpr double INELASTIC_XSEC_8TeV = 69.3; // mb
+  static constexpr double INELASTIC_XSEC_7TeV  = 68.0; // mb
+  static constexpr double INELASTIC_XSEC_8TeV  = 69.3; // mb
+  static constexpr double INELASTIC_XSEC_13TeV = 71.0; // mb
```
Where do these number for the cross-section come from ?

fwyzard commented on DQM/HLTEvF/plugins/LumiMonitor.cc:
```diff
@@ -0,0 +1,212 @@
+{  
+  
+  std::string histname, histtitle;
+
```
I would prefer not to keep a copy of the configuration as a data member, and not to extract the parameters at every run boundary.
Could you extract all the values from the configuration in the constructor, and store them in data members?

fwyzard commented on DQM/HLTEvF/plugins/LumiMonitor.cc:
```diff
@@ -0,0 +1,212 @@
+    numberOfPixelClustersVsLS_->setAxisTitle("LS",1);
+    numberOfPixelClustersVsLS_->setAxisTitle("number of pixel clusters",2);
+    
+    histname = "numberOfPixelClustersVsLumi"; histtitle = "number of pixel clusters vs lumi";
```
I would say "vs online lumi" or "vs SCAL lumi" ...

fwyzard commented on DQM/HLTEvF/plugins/LumiMonitor.cc:
```diff
@@ -0,0 +1,212 @@
+    pixelLumiVsLS_->setAxisTitle("LS",1);
+    pixelLumiVsLS_->setAxisTitle("(pixel-lumi) inst lumi E30 [Hz cm^{-2}]",2);
+    
+    histname = "pixelLumiVsLumi"; histtitle = "pixel-lumi vs lumi";
```
...especially here, because "pixel-lumi vs lumi" is somewhat confusing

fwyzard commented on DQM/HLTEvF/plugins/LumiMonitor.cc:
```diff
@@ -0,0 +1,212 @@
+					   lumiPSet.getParameter<int32_t>("nbins"), lumiPSet.getParameter<double>("xmin"), lumiPSet.getParameter<double>("xmax"),
+					   pixellumiPSet.getParameter<double>("xmin"), pixellumiPSet.getParameter<double>("xmax"));
+    pixelLumiVsLumi_->setAxisTitle("(SCAL) inst lumi E30 [Hz cm^{-2}]",1);
+    pixelLumiVsLumi_->setAxisTitle("(pixel-lumi) inst lumi E30 [Hz cm^{-2}]",2);
```
just "(pixel)" or "pixel-based" rathern than "(pixel-lumi)"

fwyzard commented on DQM/HLTEvF/plugins/LumiMonitor.cc:
```diff
@@ -0,0 +1,212 @@
+  } else {
+    scal_lumi = -1.;
+  }
+  lumiVsLS_ -> Fill(ls, scal_lumi);
```
rather than filling with "-1" if there is no value from SCAL, why don't you simply skip the Fill ?
would there be any counter indication ?

mtosi replied:
my idea is that w/ a non physical value one can easily spot possible issues w/ the input SCAL info, by distinguishing -1 from 0

fwyzard commented on DQM/HLTEvF/plugins/LumiMonitor.cc:
```diff
@@ -0,0 +1,212 @@
+    } else
+      pixel_lumi = -1.;
+
+    numberOfPixelClustersVsLS_   -> Fill(ls,       pixel_clusters);
```
same for these plots - is it better to fill with -1 for invalid pixel/scal values, or not to fill at all ?

fwyzard commented on HLTrigger/Timer/plugins/FastTimerServiceClient.cc:
```diff
@@ -223,12 +243,119 @@ FastTimerServiceClient::fillPathSummaryPlots(DQMStore::IBooker & booker, DQMStor
+  // get all MEs in the current_path
+  getter.setCurrentFolder(current_path);
+  std::vector<std::string> allmenames = getter.getMEs();
+  for ( auto m : allmenames )
```
`auto const & m` might be slightly more efficient than `auto m`

fwyzard commented on HLTrigger/Timer/plugins/FastTimerServiceClient.cc:
```diff
@@ -223,12 +243,119 @@ FastTimerServiceClient::fillPathSummaryPlots(DQMStore::IBooker & booker, DQMStor
   // Please change this to state exactly what you do use, even if it is no parameters
   edm::ParameterSetDescription desc;
   desc.addUntracked<std::string>( "dqmPath", "HLT/TimerService");
+  desc.add<bool>( "doPlotsVsSCALlumi",  true  );
```
can use consistently use `scalLumi` or `...ScalLumi` and `pixelLumi` or `...PixelLumi` for all the variables and parameter names ?

fwyzard commented on HLTrigger/Timer/plugins/FastTimerServiceClient.cc:
```diff
@@ -39,11 +41,18 @@ class FastTimerServiceClient : public DQMEDHarvester {
 
 FastTimerServiceClient::FastTimerServiceClient(edm::ParameterSet const & config) :
-  m_dqm_path( config.getUntrackedParameter<std::string>( "dqmPath" ) )
+  conf_(config)
```
same comment about `conf_` as for the other analyzer.

gregor-mittag committed:
Write compressed binaries to disk in 'MillePedeFileExtractor'.

dimattia committed:
revert

perrotta committed:
Merged 81XHLTJune15thTrain from repository cms-tsg-storm

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V362

cmsbuild committed:
Merge pull request #14953 from slava77/CMSSW_8_1_0_pre7/doubleCluVectorReserve

double the reserve for on-demand strip cluster from raw

cmsbuild committed:
Merge pull request #14942 from VinInn/ProtectTSOSinTrajFilters

protect toss in TrajFilters

cmsbuild committed:
Merge pull request #14930 from civanch/added_new_g4_stepping_verbose

Added CMS stepping verbose for SIM

davidlange6 committed:
Merge pull request #14941 from dan131riley/L1TMuonEndCap-array-bounds-fix

Fix array bounds overrun in DetectPatterns()

Martin-Grunewald commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #14938 was updated. @perrotta, @cmsbuild, @dmitrijus, @Martin-Grunewald, @fwyzard, @vanbesien, @davidlange6 can you please check and sign again.

fwyzard commented on DQM/HLTEvF/plugins/LumiMonitor.cc:
```diff
@@ -0,0 +1,246 @@
+  */
+}
+
+void LumiMonitor::getHistoPSet(edm::ParameterSet& pset, MEbinning& mebinning)
```
Can you change this to take (only) an `edm::ParameterSet const &` as input, and return a MEbinning by value ?
Something like:
```c++
MEbinning LumiMonitor::getHistoPSet(edm::ParameterSet const & pset)
{
  return MEbinning {
    pset.getParameter<int32_t>("nbins"),
    pset.getParameter<double>("xmin"),
    pset.getParameter<double>("xmax") };
}
```

This way, you can call it directly in the class initializer list.

fwyzard commented on DQM/HLTEvF/plugins/LumiMonitor.h:
```diff
@@ -0,0 +1,85 @@
+
+private:
+
+  void getHistoPSet  (edm::ParameterSet& pset, MEbinning& mebinning);
```
These two should be static .

fwyzard commented on DQM/HLTEvF/plugins/LumiMonitor.cc:
```diff
@@ -0,0 +1,246 @@
+
+LumiMonitor::LumiMonitor( const edm::ParameterSet& iConfig ) : 
+  folderName_                     ( iConfig.getParameter<std::string>("FolderName")                  )
+  , pixelClustersToken_           ( mayConsume<edmNew::DetSetVector<SiPixelCluster> >(iConfig.getParameter<edm::InputTag>("pixelClusters") ) )
```
instead of using mayConsume,  I would suggest to
  - move doPixelLumi_ (et al.) before pixelClustersToken_
  - initialise pixelClustersToken_ based on the value of doPixelLumi_:

```
pixelClustersToken_( doPixelLumi_ ? consumes<...>(...) : edm::EDGetTokenT<edmNew::DetSetVector<SiPixelCluster>> () )
```

mtosi replied:
uhmm,
but the idea is that you can monitor both at the same time ...

fwyzard commented on DQM/HLTEvF/plugins/LumiMonitor.cc:
```diff
@@ -0,0 +1,246 @@
+  , minNumberOfPixelsPerCluster_  ( iConfig.getParameter<int>        ("minNumberOfPixelsPerCluster") )
+  , minPixelClusterCharge_        ( iConfig.getParameter<double>     ("minPixelClusterCharge")       )
+{
+
```
con you move these to the initializer list, too ?

cmsbuild commented:
-1

Tested at: cb4e0ef5fa10cd80e774cf1a97bca09b7a684ee8

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14938/13637/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
10021.0 step4
<pre>runTheMatrix-results/10021.0_TenMuE_0_200+TenMuE_0_200_pythia8_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017/step4_TenMuE_0_200+TenMuE_0_200_pythia8_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017.log

</pre>1000.0 step4
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step4_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

</pre>1003.0 step3
<pre>runTheMatrix-results/1003.0_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM/step3_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM.log

</pre>1306.0 step4
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step4_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>9.0 step4
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step4_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>25.0 step4
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step4_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>1330.0 step4
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15/step4_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>10024.0 step4
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017/step4_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017.log

</pre>134.911 step4
<pre>runTheMatrix-results/134.911_RunSinglePh2015D+RunSinglePh2015D+HLTDR2_2016+RECODR2_2016reHLT+HARVESTDR2/step4_RunSinglePh2015D+RunSinglePh2015D+HLTDR2_2016+RECODR2_2016reHLT+HARVESTDR2.log

</pre>50202.0 step4
<pre>runTheMatrix-results/50202.0_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50/step4_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50.log

</pre>25202.0 step4
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step4_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>

cmsbuild committed:
Merge pull request #14891 from bsunanda/Phase2-hgx59

bsunanda:Phase2-hgx59 Add beam line for HGCal test beam studies

Martin-Grunewald committed:
Merged 81XHLTJune15thTrain from repository cms-tsg-storm

Martin-Grunewald committed:
Remove 2 paths from GRun - CMSHLT-882

Martin-Grunewald committed:
Removal of 2 paths - CMSHLT-882

gregor-mittag committed:
Keep measurements without global derivatives.

 - previously all hits without alignables defined for their subdetector
   were discarded
 - now these hits are kept by default because they are used in the
   'pede'-internaltrack fit and to define the reference frame
   -> desired default behaviour
 - a switch has been added to revert to the old behaviour
   -> useful if one wants to mask certain parts of the tracker

ianna committed:
* Clean up DDLParser

ianna committed:
* Add XercesString class

perrotta committed:
Merged 81XHLTJune15thTrain from repository cms-tsg-storm

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V365

cmsbuild commented:
Pull request #14938 was updated. @perrotta, @cmsbuild, @dmitrijus, @Martin-Grunewald, @fwyzard, @vanbesien, @davidlange6 can you please check and sign again.

fwyzard commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #14938 was updated. @perrotta, @cmsbuild, @dmitrijus, @Martin-Grunewald, @fwyzard, @vanbesien, @davidlange6 can you please check and sign again.

cmsbuild commented:
-1

Tested at: 79e55198a5311ddef29449dc6f5e50222db37149


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/353d7fd5e8877e94c2c08a281d9f54b7f6efcaf5
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14938/13645/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14938/13645/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14938/13645/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
25202.0 step2
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step2_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>50202.0 step2
<pre>runTheMatrix-results/50202.0_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50/step2_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50.log

</pre>4.22 step1
<pre>DAS Error</pre>134.911 step1
<pre>DAS Error</pre>140.53 step1
<pre>DAS Error</pre>1000.0 step1
<pre>DAS Error</pre>1001.0 step1
<pre>DAS Error</pre>1003.0 step1
<pre>DAS Error</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/353d7fd5e8877e94c2c08a281d9f54b7f6efcaf5
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14938/13645/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14938/13645/git-merge-result

dimattia committed:
Add new functionality in ConditionDBwriter.h

Martin-Grunewald committed:
Merged 81XHLTJune15thTrain from repository cms-tsg-storm

Commit:
turns out, we need to change the linkage of one more function

Martin-Grunewald committed:
Fix Datasets extraction

cmsbuild committed:
Merge pull request #14967 from slava77/CMSSW_8_0_11/checkTSOS-dyt

check if updated TSOS is good in DYT to avoid a crash

cmsbuild committed:
Merge pull request #14958 from jdefaver/clusters_store

added classes to store Phase2TrackerCluster1D

cmsbuild committed:
Merge pull request #14948 from mmusich/80X_fixAfterAbortGapHarvesting

Fix the record-tag mapping for the AAG harvesting

davidlange6 committed:
Merge pull request #14865 from kpedro88/Phase2-WF9val_rebase

Phase2 full reco workflow

mtosi commented:
I've just ran runTheMatrix -l 25202.0
I get
1 1 1 1 tests passed, 0 0 0 0 failed

Commit:
Avoid overlap seen for ME0

fwyzard commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

Martin-Grunewald committed:
Fix HLTL1TSeed accounting for prescale handling

cmsbuild commented:
Pull request #14938 was updated. @perrotta, @cmsbuild, @dmitrijus, @Martin-Grunewald, @fwyzard, @vanbesien, @davidlange6 can you please check and sign again.

fwyzard commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

Martin-Grunewald committed:
Fix string => InputTag

cmsbuild commented:
+1
Tested at: b5d43f9dfd030214085016f4216d82e16783687f
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14938/13660/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/353d7fd5e8877e94c2c08a281d9f54b7f6efcaf5
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14938/13660/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14938/13660/git-merge-result

cmsbuild commented:
Pull request #14938 was updated. @perrotta, @cmsbuild, @dmitrijus, @Martin-Grunewald, @fwyzard, @vanbesien, @davidlange6 can you please check and sign again.

mtosi commented:
before forgetting it
the HLT menu needs to be updated in order to run the new DQM source[1]
a dedicated jira ticket will be open as soon as the new code will be imported in confDB (after this PR has been merged)

[1]
process.lumiMonitor = cms.EDAnalyzer("LumiMonitor",
   FolderName                  = cms.string("HLT/LumiMonitoring"),
   doPixelLumi                 = cms.bool(False),
   useBPixLayer1               = cms.bool(False),
   minNumberOfPixelsPerCluster = cms.int32(2), # from DQM/PixelLumi/python/PixelLumiDQM_cfi.py
   minPixelClusterCharge       = cms.double(15000.),
   pixelClusters               = cms.InputTag("hltSiPixelClusters"),
   scalers                     = cms.InputTag('hltScalersRawToDigi'),
   histoPSet                   = cms.PSet(
      pixelClusterPSet            = cms.PSet(
            nbins = cms.int32 (  200  ),
            xmin  = cms.double(   -0.5),
            xmax  = cms.double(19999.5),
      ),
      lumiPSet                    = cms.PSet(
            nbins = cms.int32 ( 6500 ),
            xmin  = cms.double(        0.),
            xmax  = cms.double(13000.),
      ),
      pixellumiPSet               = cms.PSet(
            nbins = cms.int32 (300 ),
            xmin  = cms.double(  0.),
            xmax  = cms.double(  3.),
      ),
      lsPSet                      = cms.PSet(
            nbins = cms.int32 (2500 ),
      ),
   ),
)
process.hltLumiMonitoring = cms.EndPath( process.hltScalersRawToDigi + process.lumiMonitor  )

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14938/13660/summary.html

cmsbuild committed:
Merge pull request #14926 from dinardo/my-new-feature

New fits and custom labels

cmsbuild committed:
Merge pull request #14960 from ianna/ddl-parser-cleanup

DDL Parser Cleanup

davidlange6 committed:
Merge pull request #14945 from arizzi/nanProtectionPVSorter81X

add a nan protection to avoid crash in fastjet call from pvsorter (81X version)

davidlange6 committed:
Merge pull request #14719 from mandrenguyen/consumesHiGenCommon81X

Consumes migration and GenHIEventProducer forward port from 75X

Commit:
add GEMPadDigiCluster

Commit:
bugfix

smuzaffar committed:
Merge CMSSW_8_0_X into CMSSW_8_1_X.

kpedro88 committed:
add Phase2 workflows to PR tests

Commit:
moving parseHFPhase1AlgoDescription function to a more appropriate package

mbandrews committed:
Protect against neg thresholds.

Commit:
Use up-to-date HLT GT for online DQM, in DQM/Integration/python/config/FrontierCondition_GT_cfi.py.

davidlange6 committed:
Merge pull request #14951 from makortel/mtvFixStandalone

Fix to MultiTrackValidator standalone mode

davidlange6 committed:
Merge pull request #14973 from Martin-Grunewald/PrescaleAccounting81X

Fix HLTL1TSeed accounting for prescale handling (81X)

davidlange6 committed:
Merge pull request #13900 from suchandradutta/BadChanFromFED_81X

 SiStrip Bad channel estimation from FED error

davidlange6 committed:
Merge pull request #14623 from mtosi/fixTrackingPlot

add DCA plots in tracking@HLT DQM

VinInn committed:
cleanup MeasurementTrackerEvent

VinInn committed:
Update MeasurementTrackerEvent.h

just doc

Commit:
renamig a file

Commit:
updating the guard name

davidlange6 committed:
Merge pull request #14726 from leaca/LC_BPIXPH1

Updated Phase 1 BPIX Geometry

perrotta committed:
Merged 81XHLTJune15thTrain from repository cms-tsg-storm

perrotta committed:
Update the HLT menu up to /dev/CMSSW_8_0_0/HLT/V370

Commit:
DataFormats/HcalRecHit: Add HFQIE10Info::N_RAW_MAX to HFQIE10Info.cc

The patch resolves issue with Clang compiler.
N3690 (should be C++11 standard) and latest draft N4567, 9.4.2/3

...
The member shall still be defined in a namespace scope if it
is odr-used (3.2) in the program and the namespace scope definition
shall not contain an initializer.
9.4.2/4 talks about how const static data members are being handled.

Also standard says that no diagnostic is required by compiler.

    src/DataFormats/HcalRecHit/src/HFQIE10Info.cc:(.text+0x7c): undefined
    reference to FQIE10Info::N_RAW_MAX'

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Commit:
Changed errors for hit and vtx counters

Commit:
Update Vx3DHLTAnalyzer.cc

hroskes committed:
don't override python builtin

hroskes committed:
Enable auto: global tags

bapavlov committed:
rpc and irpc code factorization

hroskes committed:
BPIX and FPIX surface deformation plots for phase1

hroskes committed:
comment drawing the title
(can put it back for individual plots when necessary)

gartung committed:
Fix  this 'if' clause does not guard... [-Wmisleading-indentation]

Commit:
add GEMPadDigiCluster

Commit:
add GEMPadDigiClusterProducer

Commit:
keep GEMPadDigiClusters

gartung committed:
Fix  this 'if' clause does not guard... [-Wmisleading-indentation]

gartung committed:
Fix  this 'if' clause does not guard... [-Wmisleading-indentation]

bapavlov committed:
fix in muonRPCDigis_cfi.py

gartung committed:
Fix  this 'if' clause does not guard... [-Wmisleading-indentation]

gartung committed:
Fix  this 'if' clause does not guard... [-Wmisleading-indentation]

gartung committed:
Fix  this 'if' clause does not guard... [-Wmisleading-indentation]

gartung committed:
Fix  this 'if' clause does not guard... [-Wmisleading-indentation]

gartung committed:
Fix  this 'if' clause does not guard... [-Wmisleading-indentation]

gartung committed:
Fix  this 'if' clause does not guard... [-Wmisleading-indentation]

gartung committed:
Fix  this 'if' clause does not guard... [-Wmisleading-indentation]

gartung committed:
how the compiler sees it

gartung committed:
how the compiler sees it

gartung committed:
this is how the compiler sees it

gartung committed:
this is how the compiler sees it

cmsbuild committed:
Merge pull request #14983 from VinInn/CleanupMeasurementTrackerEvent

cleanup MeasurementTrackerEvent

cmsbuild committed:
Merge pull request #14980 from diguida/use_updated_HLT_GT_DQM_81X

Use up-to-date HLT GT for online DQM (81X)

cmsbuild committed:
Merge pull request #14979 from mbandrews/NegThr81X

Protect against negative occupancy thresholds

cmsbuild committed:
Merge pull request #14971 from bsunanda/Phase2-gem17

bsunanda:Phase2-gem17 Avoid overlap seen for ME0

cmsbuild committed:
Merge pull request #14961 from ghellwig/reduce-PCL-disk-space_81X

Reduce PCL disk space 81X

Martin-Grunewald commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 055e73d637c31a8aacf9649c66ab5f61daa45aa2
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14938/13720/summary.html

gartung committed:
fix warning with curly braces as author intended

gartung committed:
tab to spaces

gartung committed:
tab to spaces

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14938/13720/summary.html

jhgoh committed:
remove SETMuons from the DT DQM

gartung committed:
probably what author intended

hroskes committed:
Fix * syntax in [validation] section

gartung committed:
probably what the author intended

mtosi committed:
Merged refs/pull/14857/head from repository cms-sw

Conflicts:
	DQM/TrackingMonitorSource/python/TrackingSourceConfig_Tier0_cff.py

gartung committed:
probably what author intended

kpedro88 committed:
add 2017 WFs to limited

gartung committed:
formatting fit

gartung committed:
formatting fix

gartung committed:
formatting fix

davidlange6 committed:
Merge pull request #15011 from fwyzard/hltDiff_fix_process_name_in_candidates

hltDiff: fix comparison of trigger candidates with different process name

davidlange6 committed:
Merge pull request #14986 from silviodonato/HLTBTVRelValFix

Fixes in HLT BTV release validation DQM

cmsbuild committed:
Merge pull request #14220 from uhussain/CMSSW_8_1_X

Fixed static analyzer warning in PFCandidatePhotonExtra.h, CSCDCCUnpacker.cc ,CastorUnpacker.cc, DTDigiToRaw.cc and CamacTBDataFormatter.cc

cmsbuild committed:
Merge pull request #15013 from cms-l1t-offline/pr80x-fix-L1EG-relaxID-forHigPt

L1T EG relax of ID cuts (including HoE) for high pt objects above tower saturation

cmsbuild committed:
Merge pull request #14992 from cms-l1t-offline/pr80x-fix-L1TSeed-crashHLT-correlations

PR 81X Fix sporadic crashes in HLT when Seeding with correlations.

davidlange6 committed:
Merge pull request #14966 from dimattia/multiRunHarvesting_for_8_0_X

Multi run harvesting for 8 1 x

davidlange6 committed:
Merge pull request #14684 from shervin86/ALCARECOEcalTrg

Alcareco ecal trg

cmsbuild committed:
Merge pull request #14611 from heppye/patch35

add RelVal fragmentation files for BPH

gartung committed:
formatting fix

gartung committed:
formatting fix

gartung committed:
This is probably what the author intended. See similar statements lower in the code.

gartung committed:
formatting fix

gartung committed:
formatting fix

threus committed:
add init for fedIdVsApvId_

gartung committed:
formatting fix

gartung committed:
bug fix

gartung committed:
formatting change

gartung committed:
formatting fix

gartung committed:
formatting fix

gartung committed:
forgot to add these changes

gartung committed:
formatting fix

mtosi committed:
fix HLT tracking client

cmsbuild committed:
Merge pull request #14999 from gartung/Validation-RecoParticleFlow-fix-indent-not-guard-warn

Validation/RecoParticleFlow : gcc 6.0 misleading-indentation warning flags potential bug; with bug fix

cmsbuild committed:
Merge pull request #14957 from igv4321/timeshift_ns_hbheho

Changing linkage of the "timeshift_ns_hbheho" function to external

cmsbuild committed:
Merge pull request #15004 from gartung/SimG4CMS-FP420-fix-indent-not-guard-warn

SimG4CMS/FP420 : gcc 6.0 misleading-indentation warning flags potential bug; with formatting fix

cmsbuild committed:
Merge pull request #15006 from gartung/SimG4CMS-Forward-fix-indent-not-guard-warn

SimG4CMS/Forward : gcc 6.0 misleading-indentation warning flags potential bug; with bug fix

cmsbuild committed:
Merge pull request #15005 from gartung/SimG4CMS-Calo-fix-indent-not-guard-warn

SimG4CMS/Calo : gcc 6.0 misleading-indentation warning flags potential bug; with formatting fix

cmsbuild committed:
Merge pull request #15041 from gartung/HLTrigger-Tools-fix-indent-not-guard

HLTrigger/Tools : gcc 6.0 misleading-indentation warning flags potential bug; with formatting fix

mtosi committed:
add EGM tracking@HLT monitoring

mtosi committed:
add EGM tracking@HLT monitoring

Martin-Grunewald committed:
Merged 81XHLTJune15thTrain from repository cms-tsg-storm

cmsbuild committed:
Merge pull request #14920 from igv4321/phase1-hf-timingfix

Mostly changes related to handling TDC info

cmsbuild committed:
Merge pull request #15001 from dildick/from-CMSSW_8_1_0_pre7-GEMPadDigiClusterizer

New feature in GEM simulation: GEMPadDigiCluster

jhgoh committed:
remove calomuon producer from the HI reco

Martin-Grunewald committed:
CMSHLT-885

Dr15Jones committed:
Improve efficiency of trigTools::setFiltersObjPasses

The function was creating a edm::InputTag, converting that to a string
then passing the string to a function which needs an edm::InputTag so
the compiler created yet another edm::InputTag. Now the code just
directly passes the edm::InputTag to the function.

This code was brought to my attention because running this code
using multiple threads has periodically crashed at this line of code.
The problem is not understood and it is unlikely that this change
fixes the underlying problem.

BetterWang committed:
[!] bug fix

(cherry picked from commit eecbc979b4b833fe4b4fedbc51c6a6202c9594e8)

gartung committed:
add curly braces

Commit:
Correct several scripts to enable them useful yet again

mandrenguyen committed:
synch heavy-ion customizations with 80X

Commit:
Correct BH geometry...

mandrenguyen committed:
fix indent error

perrotta committed:
Merged 81XHLTJune15thTrain from repository cms-tsg-storm

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V381, i.e. frozen/online versions 2.2

Commit:
Take suggestion from Vladimir to resolve namespace issue

Commit:
Change Physics List option as suggested by Vladimir

gartung committed:
gcc 6.0 misleading-indent warning flagged these. I think the author indended
to fill the historgram with the value just set. I added the curly braces
to do that. Without them the last value of fullType set would be added to
all of the histograms.

gartung committed:
These are formatting changes to fix gcc 6.0 misleading-indent warning.

gartung committed:
Fixes for bugs flagges by gcc 6.0 misleading-indentation warning.

cmsbuild committed:
Merge pull request #15023 from gartung/Calibration-IsolatedParticles-fix-indent-not-guard

Calibration/IsolatedParticles : gcc 6.0 misleading-indentation warning flags potential bug; with formatting fix

cmsbuild committed:
Merge pull request #15021 from gartung/CalibCalorimetry-HcalAlgos-fix-indent-not-guard

CalibCalorimetry/HcalAlgos :  gcc 6.0 misleading-indentation warning flags potential bug; with formatting fix

cmsbuild committed:
Merge pull request #15019 from gartung/CalibCalorimetry-EcalLaserAnalyzer-fix-indent-not-guard

CalibCalorimetry/EcalLaserAnalyzer : gcc 6.0 misleading-indentation warning flags potential bug; with bug fix

cmsbuild committed:
Merge pull request #15018 from gartung/Alignment-TrackerAlignment-fix-indent-not-guard-warn

Alignment/TrackerAlignment : gcc 6.0 misleading-indentation warning flags potential bug; with formatting fix

cmsbuild committed:
Merge pull request #15069 from gartung/Fireworks-fix-ident-not-guards-warn

Fireworks : Fixes for bugs flagged by gcc 6.0 misleading-indentation warning.

davidlange6 committed:
Merge pull request #15060 from cms-tsg-storm/81XHLTJune15thTrain

81X HLT June15th train

smuzaffar committed:
Merge CMSSW_8_0_X into CMSSW_8_1_X.

bartosik-hep committed:
Merge remote-tracking branch 'my-cmssw/CMSSW_7_6_X_hltDiff_json' into CMSSW_8_1_X_hltDiff_json

Conflicts:
	HLTrigger/Tools/bin/hltDiff.cc

bartosik-hep committed:
Fixed a bug from the 7_6_X merge conflict

davidlange6 committed:
Merge pull request #14950 from iahmad-khan/tinyxml-fixed

fixes tinyxml dependencies and uses tinyxml as external

davidlange6 committed:
Merge pull request #15068 from gartung/DQMServices-Core-fix-indent-not-guard-warn

DQMServices/Core : gcc 6.0 misleading-indentation warning flags potential bug; with formatting fix ???

cmsbuild committed:
Merge pull request #15007 from gartung/RecoLocalMuon-CSCSegment-fix-indent-not-guard-warn

RecoLocalMuon/CSCSegment : gcc 6.0 misleading-indentation warning flags potential bug ; with formating fix

mtosi committed:
add V0 monitoring

safarzad committed:
update Higgs and SUSY DQM

cmsbuild committed:
Merge CMSSW_8_0_X into CMSSW_8_1_X.

mtosi committed:
fix mass axis

mtosi committed:
cleaning

Commit:
Update the TB simulation application with beam line simulation

smuzaffar committed:
Merge CMSSW_8_0_X into CMSSW_8_1_X.

gartung committed:
DQM/HcalMonitorClient : Formatting fix for gcc 6.0 missleading-indentation warning.

mtosi committed:
fix pdgID

mtosi committed:
fix mass

cmsbuild committed:
Merge pull request #15093 from bsunanda/Phase2-hgx61

bsunanda:Phase2-hgx61 Update the TB simulation application with beam line simulation

cmsbuild committed:
Merge pull request #15054 from bsunanda/Run2-hcx85

bsunanda:Run2-hcx85 Correct several scripts to enable them useful yet again

cmsbuild committed:
Merge pull request #14977 from kpedro88/Phase2-WF10

add Phase2 workflows to PR tests

mtosi committed:
fix

mtosi committed:
fix

Martin-Grunewald commented:
+1

davidlange6 committed:
Merge pull request #14997 from hroskes/alignment-validation-phase1-81X

Fix DMR validation for phase 1

davidlange6 committed:
Merge pull request #15016 from slava77/CMSSW_8_0_12/sign733/LogErrorsA

suppress LogError for somewhat clearly bad input of TrajectoryStateCombiner

davidlange6 committed:
Merge pull request #15094 from gartung/DQM-HcalMonitor-client-fix-indent-not-guard-warn

DQM/HcalMonitorClient : Formatting fix for gcc 6.0 misleading-indentation warning.

gregor-mittag committed:
Make pointer to 'AlignmentLevel' std::unique_ptr.

Commit:
correct one namespace issue

civanch committed:
fixed static analiser warnings

cmsbuild committed:
Merge pull request #14988 from davidlt/fix-N_RAW_MAX-clang

DataFormats/HcalRecHit: Add HFQIE10Info::N_RAW_MAX to HFQIE10Info.cc

cmsbuild committed:
Merge pull request #15056 from mandrenguyen/ppRecoHIcustomizations81X

synch heavy-ion customizations with 80X

cmsbuild committed:
Merge pull request #15038 from gartung/EventFilter-SiPixelRawToDigi-fix-indent-not-guard

EventFilter/SiPixelRawToDigi : formatting fix for gcc 6.0 misleading-indentation warning

cmsbuild committed:
Merge pull request #15036 from gartung/DataFormats-TrackerRecHit2D-fix-indent-not-guard

DataFormats/TrackerRecHit2D : formatting fix for gcc 6.0 misleading-indentation warning

cmsbuild committed:
Merge pull request #15026 from gartung/CommonTools-Utils-fix-indent-not-guard

CommonTools/Utils : formatting fix for gcc 6.0 misleading-indentation warning

cmsbuild committed:
Merge pull request #15009 from gartung/MagneticField-Interpolation-fix-indent-not-guard-warn

MagneticField/Interpolation : formatting fix for gcc 6.0 misleading-indentation warning

cmsbuild committed:
Merge pull request #15008 from gartung/RecoLocalMuon-CSCValidation-fix-indent-not-guard-warn

RecoLocalMuon/CSCValidation: formatting fix for gcc 6.0 misleading-indentation warning

cmsbuild committed:
Merge pull request #14857 from mtosi/addHitEffVsLUMI

 add track hit eff monitoring vs inst. lumi

davidlange6 committed:
Merge pull request #15100 from BetterWang/ClusterShapeTrackFilter_81X

Protection against empty globalDirs

Dr15Jones committed:
Fix unary operator handling in FormulaEvalutor

The code now properly handles '-a*b*c' where the code had dropped
'c' because of the way operator precedence is handled.

ianna committed:
New tracker reco material for 2017 scenario

argiro committed:
add option to switch off time calibration

Martin-Grunewald commented:
@dmitrijus, @vanbesien 
please sign or comment otherwise

cmsbuild committed:
Merge pull request #15098 from civanch/fix_static_analyser_warn3

fixed static analiser warnings3

cmsbuild committed:
Merge pull request #15083 from bartosik-hep/CMSSW_8_1_X_hltDiff_json

Extended hltDiff with CVS/JSON/ROOT output formats [8_1_X]

davidlange6 committed:
Merge pull request #15055 from bsunanda/Phase2-hgx60

bsunanda:Phase2-hgx60 Correct BH geometry...

cmsbuild committed:
Merge pull request #15040 from gartung/Geometry-EcalCommonData-fix-indent-not-guard

Geometry/EcalCommonData : formatting fix for gcc 6.0 misleading-indentation warning

cmsbuild committed:
Merge pull request #14969 from fioriNTU/AddLayNum

Adding Layer number to bad module list

leggat committed:
Implementing pixel cluster trend plots

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_0_X IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @degano, @smuzaffar

cmsbuild committed:
Merge pull request #15108 from kirschen/addPFMET190PathDQMPlots

add Type1-corrected PFMET path to SUSY HLT DQM monitoring

cmsbuild committed:
Merge pull request #15090 from mtosi/V0monitoring

add V0 monitoring

cmsbuild committed:
Merge pull request #15088 from safarzad/DQMplots

update Higgs and SUSY DQM

cmsbuild committed:
Merge pull request #14769 from bapavlov/RPC_and_IRPC_digitization_PhaseII

Rpc and irpc digitization phase ii

cmsbuild committed:
Merge pull request #14793 from threus/federrors_81

FED Errors vs FED ID 2D plot

cmsbuild committed:
Merge pull request #15033 from gartung/DQM-L1TMonitor-fix-indent-not-guard

DQM/L1TMonitor : gcc 6.0 misleading-indentation warning flags potential bug; with formatting fix

cmsbuild committed:
Merge pull request #15032 from gartung/DQM-HcalMonitorTasks-fix-indent-not-guard

DQM/HcalMonitorTasks : gcc 6.0 misleading-indentation warning flags potential bug; with formatting fix

cmsbuild committed:
Merge pull request #15030 from gartung/DQM-CastorMonitor-fix-indent-not-guard

DQM/CastorMonitor : formatting fix for gcc 6.0 misleading-indentation warning

cmsbuild committed:
Merge pull request #15035 from gartung/DQM-SiStripCommissioningDbClients-fix-indent-not-guard

DQM/SiStripCommissioningDbClients fix for bug flagged by gcc 6.0 misleading-indentation warning

cmsbuild committed:
Merge pull request #15067 from gartung/DQM-SiPixelMonitorRawData-fix-indent-not-guard

DQM/SiPixelMonitorRawData fix for bug flagged by gcc 6.0 missleading-indent warning

cmsbuild committed:
Merge pull request #15051 from Dr15Jones/improve_setFiltersObjPasses

Improve efficiency of trigTools::setFiltersObjPasses

cmsbuild committed:
Merge pull request #14747 from jhgoh/RemoveSETMuonCaloMuon

Disable SETMuon and caloMuon from the standard reconstruction

Commit:
Move the BH geometry to point the center of the cell

ianna committed:
* Allow compilation with EDM_ML_DEBUG
* Silence non-debug printout

ggovi committed:
Various improvements in the conddb command line tools

fioriNTU committed:
FixAxisLimits

ggovi committed:
Fix for o2o job enabling/disabling

cmsbuild committed:
Merge pull request #14990 from dinardo/my-new-feature_81

Changed errors for hit and vtx counters

cmsbuild committed:
Merge pull request #15000 from gartung/CaloOnlineTools-HcalOnlineDb-fix-indent-not-guard-warn

CaloOnlineTools/HcalOnlineDb : formatting fix for gcc 6.0 misleading-indentation warning

cmsbuild committed:
Merge pull request #15028 from gartung/CondFormats-TotemReadoutObjects-fix-indent-not-guard

CondFormats/TotemReadoutObjects : fix for bug flagged by gcc 6.0 misleading-indentation warning

cmsbuild committed:
Merge pull request #15027 from gartung/CondCore-Utilities-fix-indent-not-guard

CondCore/Utilities : formatting fix for gcc 6.0 misleading-indentation warning

cmsbuild committed:
Merge pull request #15045 from mtosi/DQMEGMTracking@HLT

DQM egm tracking@HLT

cmsbuild committed:
Merge pull request #15132 from ggovi/condcore-utilities-improvements-0

Various improvements in the conddb command line tools

davidlange6 committed:
Merge pull request #15106 from Dr15Jones/bugFixFormulaEvaluator

Fix unary operator handling in FormulaEvalutor

jonuuukas committed:
Changing quotes of AlCa matrix to be JSON valid

jonuuukas committed:
fixing the line 20 with adding '+EcalTrg'

ebrondol committed:
removing

mmusich committed:
Update Global Tags: new JEC, introding HcalFrontEndMapRcd, updated Phase-I Pixel Geometry

Commit:
Control path for DisplacedJets

smuzaffar committed:
Merge CMSSW_8_0_X into CMSSW_8_1_X.

davidlange6 committed:
Merge pull request #15143 from jruizvar/disJet81X

Control path for DisplacedJets (81X)

davidlange6 committed:
Merge pull request #15141 from mmusich/updateGlobalTags_81X_Jul7_2016

Update Global Tags: new JEC, introducing HcalFrontEndMapRcd, updated Phase-I Pixel Geometry

davidlange6 committed:
Merge pull request #15139 from jonuuukas/where-magic-happens

Changing quotes of AlCa matrix to be JSON valid

davidlange6 committed:
Merge pull request #15138 from ggovi/condcore-utilities-o2o-fix0

Fix for o2o job enabling/disabling

davidlange6 committed:
Merge pull request #15134 from fioriNTU/FixAxisLimits

FixAxisLimits

davidlange6 committed:
Merge pull request #15126 from bsunanda/Phase2-hgx62

bsunanda:Phase2-hgx62 Move the BH geometry to point the center of the cell

davidlange6 committed:
Merge pull request #15121 from leggat/pixel-cluster-trend-plots

Implementing pixel cluster trend plots

davidlange6 committed:
Merge pull request #15113 from argiro/ecalOptionNoTimeCalib

add option to switch off time calibration 81x

davidlange6 committed:
Merge pull request #15110 from ianna/tk-reco-material-2017

Tracker Reco Material for 2017 Scenario

davidlange6 committed:
Merge pull request #15097 from ghellwig/phase1_alignment-updates_81X

Phase-1 alignment updates

davidlange6 committed:
Merge pull request #15024 from gartung/Calibration-Tools-fix-indent-not-guard

Calibration/Tools : formatting fix for gcc 6.0 misleading-indentation warning

davidlange6 committed:
Merge pull request #15003 from gartung/RecoTracker-MeasurementDet-fix-indent-not-guard-warn

RecoTracker/MeasurementDet : bug fix for gcc 6.0 misleading-indentation warning

davidlange6 committed:
Merge pull request #14994 from ebrondol/fixDuplicate_ph2Tracking

Reducing duplicates in Phase2 tracking reconstruction (with some minor effects on phase0/1 tracking)

davidlange6 committed:
Merge pull request #14843 from matt-komm/hitmerging-master

New FastSim Hit Reconstruction Emulation

davidlange6 committed:
Merge pull request #15128 from ianna/fix-edm-ml-debug-compilation

Fix Compilation Errors with EDM_ML_DEBUG Defined

cmsbuild commented:
Pull request #14938 was updated. @ghellwig, @cerminar, @cmsbuild, @rekovic, @franzoni, @mmusich, @mulhearn, @davidlange6 can you please check and sign again.

cmsbuild commented:
Pull request #14938 was updated. @ghellwig, @cerminar, @cmsbuild, @rekovic, @franzoni, @mmusich, @mulhearn, @davidlange6 can you please check and sign again.

mtosi commented:
sorry the confusion (and mess as well)

I think I have to close and open a new PR
because re-basing the timingVSlumi branch for 81x
this PR gets spoiled :(

do you agree ?
I mean, are there other smarter ways ?
thanks

@slava77 @Dr15Jones @davidlange6

dmitrijus commented:
You don't have to create a new pull request - just do a forced push (git push -f) after a rebase. 
But whichever you decide, please close this or the other pull request.

mtosi commented:
as said I had issue w/ the synch of the branch in 81x and 80x
=> I opened the PR https://github.com/cms-sw/cmssw/pull/15169
=> close this one
sorry the confusion and thanks
